src/dmd/cparse.d

```diff
@@ -0,0 +1,3178 @@
+
+        case TOK.int64Literal:
+            e = new AST.IntegerExp(loc, token.intvalue, AST.Type.tint64);
+            nextToken();
+            break;
+
+        case TOK.uns64Literal:
+            e = new AST.IntegerExp(loc, token.unsvalue, AST.Type.tuns64);
+            nextToken();
+            break;
+
+        case TOK.float32Literal:
+            e = new AST.RealExp(loc, token.floatvalue, AST.Type.tfloat32);
+            nextToken();
+            break;
+
+        case TOK.float64Literal:
+            e = new AST.RealExp(loc, token.floatvalue, AST.Type.tfloat64);
+            nextToken();
+            break;
+
+        case TOK.float80Literal:
+            e = new AST.RealExp(loc, token.floatvalue, AST.Type.tfloat80);
+            nextToken();
+            break;
+
+        case TOK.imaginary32Literal:
+            e = new AST.RealExp(loc, token.floatvalue, AST.Type.timaginary32);
+            nextToken();
+            break;
+
+        case TOK.imaginary64Literal:
+            e = new AST.RealExp(loc, token.floatvalue, AST.Type.timaginary64);
+            nextToken();
+            break;
+
+        case TOK.imaginary80Literal:
+            e = new AST.RealExp(loc, token.floatvalue, AST.Type.timaginary80);
+            nextToken();
+            break;
+
+        case TOK.functionString:
+            e = new AST.FuncInitExp(loc);
+            nextToken();
+            break;
+
+        case TOK.charLiteral:
+            e = new AST.IntegerExp(loc, token.unsvalue, AST.Type.tchar);
+            nextToken();
+            break;
+
+        case TOK.wchar_tLiteral:
+            e = new AST.IntegerExp(loc, token.unsvalue, AST.Type.twchar); // correct to twchar_t in semantic()
+            nextToken();
+            break;
+
+        case TOK.string_:
+        {
+            // cat adjacent strings
+            auto s = token.ustring;
+            auto len = token.len;
+            auto postfix = token.postfix;
+            while (1)
+            {
+                nextToken();
+                if (token.value == TOK.string_)
+                {
+                    if (token.postfix)
+                    {
+                        if (token.postfix != postfix)
+                            error("mismatched string literal postfixes `'%c'` and `'%c'`", postfix, token.postfix);
+                        postfix = token.postfix;
+                    }
+
+                    const len1 = len;
+                    const len2 = token.len;
+                    len = len1 + len2;
+                    auto s2 = cast(char*)mem.xmalloc_noscan(len * char.sizeof);
+                    memcpy(s2, s, len1 * char.sizeof);
+                    memcpy(s2 + len1, token.ustring, len2 * char.sizeof);
+                    s = s2;
+                }
+                else
+                    break;
+            }
+            e = new AST.StringExp(loc, s[0 .. len], len, 1, postfix);
+            break;
+        }
+
+        case TOK._Generic:
+            assert(0);
```

Comment:
Preferably we'd have unimplemented errors instead of an ICE in the parser.

Reply:
added message